### PR TITLE
Added a mechanism and cops to check for undesirable migration changes

### LINKED ISSUE #26: Write Rake CI task to warn on committed migrations with no structure dump
As a developer
I want to be reminded through the CI process when a migration is committed with no structure dump
So that I will be nudged towards doing so.

### PR DESCRIPTION
**Pull request by timgentry**
This PR resolves #26

### CODE REVIEW

File: lib/ndr_dev_support/rake_ci/commit_cop/renamed_migration.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+module NdrDevSupport
+  module RakeCI
+    module CommitCop
+      # This cop checks for renamed migrations allowing for unscoped to scoped name changes
+      class RenamedMigration
+        include Deputisable
+
+        def check(changes)
+          renamed_migrations = changes[:renamed].select(&migration_file?)
+          return if renamed_migrations.empty?
```
Comment (joshpencheon) [inline lib/ndr_dev_support/rake_ci/commit_cop/renamed_migration.rb]: I think this is redundant, given the next guard clause: `[].all? #=> true`

File: lib/ndr_dev_support/rake_ci/commit_cop/renamed_migration.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          return if renamed_migrations.all? do |old_file, new_file|
+            unscoped_migration_file?.call(old_file) && !unscoped_migration_file?.call(new_file)
```
Comment (joshpencheon) [inline lib/ndr_dev_support/rake_ci/commit_cop/renamed_migration.rb]: The `SCHEMA_VERSIONS` table only tracks the timestamp as the migration identifier, so this check could generate false positives; it wouldn't be problematic to rename the Ruby classes, e.g. in the case of mis-spelling or a need for additional clarity down the line.

File: lib/ndr_dev_support/rake_ci/commit_cop.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+require_relative 'commit_cop/modified_migration'
+require_relative 'commit_cop/renamed_migration'
+
+module NdrDevSupport
+  module RakeCI
+    # This module encapsulates commit cop logic
+    module CommitCop
+      # This defines the regular expression that identifies the path to migration files.
+      mattr_accessor :migration_path_pattern
+      self.migration_path_pattern = %r{\Adb/migrate/}
```
[the remark this answers is not in the document]
Comment (timgentry) [reply lib/ndr_dev_support/rake_ci/commit_cop.rb]: I'm on it

**Comment by timgentry** [comment]
@joshpencheon this should be mergeable now.

**Comment by joshpencheon** [comment]
Cool, I didn't know `start_with?` could take multiple candidates:
```ruby
'hello'.start_with?('hello', 'world') #=> true
```

**Pull request merged**